torch/_inductor/fx_passes/b2b_gemm.py

```diff
@@ -245,23 +255,105 @@ def load_ratio_right(
 
 
 # the block sizes are limited by hardware (the shared memory)
-# in theory all numbers are independent; e.g. BLOCK_SIZE_M is not necessarily equal to BLOCK_SIZE_O
+# intuitively, the optimization works when the intermediate matrix is large
+# and we assign large block sizes to large dimensions
 b2b_gemm_configs = [
     {
-        "BLOCK_SIZE_M": m,
-        "BLOCK_SIZE_N": n,
-        "BLOCK_SIZE_O": m,
-        "BLOCK_SIZE_P": n,
-        "num_stages": s,
-        "num_warps": 2 * s,
-    }
-    for (m, n, s) in {  # deduplicate
-        (m, n, s)
-        for m in [16, 32, 64, 128]
-        for n in [16, 32, 64, 128]
-        for s in ([2] if (m == 128 or n == 128) else [2, 4])  # don't be too large
-        if not (m == n == 128)  # don't be too large
-    }
+        "BLOCK_SIZE_M": 128,
```

Comment:
Do we need to be careful about the number of different configs we use? I think I've heard @eellison describe how it can be a problem for compile time, but if we intend this will only be enabled with max-autotune, maybe it's less of a problem?

Reply:
I'm roughly following the numbers of configs here (https://github.com/pytorch/pytorch/blob/main/torch/_inductor/kernel/mm_common.py#L120-L141). Perhaps I can also try to benchmark the compilation time.

Reply:
Yea, it's okay to lean on extra configs initially.